src/core/unit.c

```diff
@@ -1281,6 +1281,12 @@ int unit_add_exec_dependencies(Unit *u, ExecContext *c) {
                 r = unit_add_dependency_by_name(u, UNIT_AFTER, SPECIAL_TMPFILES_SETUP_SERVICE, true, UNIT_DEPENDENCY_FILE);
                 if (r < 0)
                         return r;
+        } else if (c->private_tmp == PRIVATE_TMP_DISCONNECTED) {
+                /* Even if PrivateTmp=disconnected, we still require /var/tmp/ mountpoint to be present,
+                 * i.e. /var/ needs to be mounted. */
+                r = unit_add_mounts_for(u, "/var", UNIT_DEPENDENCY_FILE, UNIT_MOUNT_WANTS);
```

Comment:
This doesn't look right, as the entire point was to disconnect such services from these mount points, so it's going to break use cases like softreboot

Reply:
> so it's going to break use cases like softreboot

Hmm? Can you elaborate? Note that the added dep is weak, i.e. unmount would not be propagated. We certainly shouldn't create anything under `/var/` on rootfs if a separate `/var/` mount is desired.

Reply:
Services that are disconnected from the rootfs (because they use RootImage= and friends) will be tied to it via the new dependency, which is not right - again that's why I did the original change, to ensure such services would avoid these dependencies and ordering and implied conflicts and so on

Reply:
Wouldn't also just `After=` suffice? The unit will already be loaded since the generators run before starting units, and the start job queued, so this will then just change the ordering?

Reply:
> Services that are disconnected from the rootfs (because they use RootImage= and friends) will be tied to it via the new dependency

Then again, this is the very nature of Linux mount namespacing, that the mountpoint must exist. Otherwise you're breaking use case of ro `/` and separate read write `/var/`...

Reply:
Please also add `After=`.

Reply:
> Please also add `After=`.

That's implied by WantMountsFor=.

Reply:
> > Please also add `After=`.
> 
> That's implied by WantsMountsFor=.

Oh, I indeed. Sorry for the noise. LGTM.